packages/sdks/src/components/block/components/component-ref/component-ref.lite.tsx

```diff
@@ -28,13 +29,52 @@ useMetadata({
  */
 export default function ComponentRef(props: ComponentProps) {
   const state = useStore({
+    shouldUpdate: false,
     Wrapper: props.isInteractive
       ? useTarget({
           vue: wrapComponentRef(InteractiveElement),
           default: InteractiveElement,
         })
       : props.componentRef,
   });
+
+  onUpdate(() => {
+    useTarget({
+      angular: () => {
+        if (
+          // @ts-expect-error - 'changes' comes from Angular's ngOnChanges hook
+          !changes['blockChildren']?.isFirstChange() &&
+          // @ts-expect-error - 'changes' comes from Angular's ngOnChanges hook
+          JSON.stringify(changes['blockChildren']?.previousValue) !==
```

Comment:
isn't this a massive cost, stringifying the entire block children on _every_ `ngOnChanges`?

Reply:
any way it can be avoided?

Reply:
yup, updated the logic to only check for changes in blockChildren, had to add the previousValue && currVal !== 0 as getter will send a new empty array every time the change detection runs

loom of the approach:
https://www.loom.com/share/5c3c3952775442099ef5dc6ba626d340